swap/src/cli/command.rs

```diff
@@ -22,10 +22,10 @@ pub struct Arguments {
 pub enum Command {
     BuyXmr {
         #[structopt(long = "connect-peer-id")]
-        alice_peer_id: PeerId,
+        alice_peer_id: Option<PeerId>,
 
         #[structopt(long = "connect-addr")]
-        alice_addr: Multiaddr,
+        alice_addr: Option<Multiaddr>,
```

Comment:
Structopt supports default values, that might make the whole thing more concise because you don't need to do `map_or` everywhere. https://docs.rs/structopt/0.3.21/structopt/index.html#how-to-derivestructopt